### PR TITLE
Add font sizes

### LINKED ISSUE #36: Add font size presets
Figma link: https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3112-304&t=qC1CDiog2u4tpIsg-4

In the Figma file I am seeing the following sizes:

| Headings | Paragraphs  |
|--------------------|----------------------------------|
| H1: 70px    |   -             |
| H2 48px    |   -             |
| H3 38px    |  Extra large 38px     |
| H4 28px    |  Large  28px       |
| H5  24px   |    -            |
| H6 22px    |  Regular 22px    |
| -    |  Small 18px    |
| -    |  Extra Small 14px    |
| -    |  Tiny 12px    |

Which gives us 9 font sizes. My recommendation would be to:
- Not use 12 and 14 because they are difficult to read (arguably, depending on the font family).
- Use 24px for the H5 heading as a one-off size under styles > elements > heading similar to TT4:
https://core.trac.wordpress.org/browser/trunk/src/wp-content/themes/twentytwentyfour/theme.json#L860

- Possibly also use either 28px or 38px as a one-off size under styles > elements > heading.
 I do not know how widely used these sizes are in the design, but 70 and 48 are common.
Having a gap from 22 to 48 is rather large, so my preference would be to remove 28px from the scale.

Small 18px
Regular 22px ( The default body text size.)
Large 28px or 38px
Extra large 48px
Extra Extra large 70px

### PR DESCRIPTION
**Pull request by carolinan**
**Description**
A first go at adding the font sizes.
Closes https://github.com/WordPress/twentytwentyfive/issues/36

**Screenshot**
font-size-control-tt5

**Testing Instructions**
1. Apply the PR
2. Confirm that the sizes are applied correctly to the body, headings, and the updated blocks, for example
Site title, tagline, search, code, post date.

### CODE REVIEW

File: theme.json [hunk 1]
```diff
@@ -97,6 +97,63 @@
...
+					"slug": "x-large"
+				},
+				{
+					"fluid": {
+						"max": "58px",
+						"min": "42px"
+					},
+					"name": "Extra Extra Large",
+					"size": "42px",
+					"slug": "xx-large"
```
Comment (carolinan) [inline theme.json:153]: I renamed this from "huge".

File: theme.json [hunk 2]
```diff
@@ -97,6 +97,63 @@
...
+				},
+				{
+					"fluid": {
+						"max": "18px",
+						"min": "18px"
+					},
+					"name": "Small",
+					"size": "1rem",
+					"slug": "small"
+				},
```
Comment (carolinan) [inline theme.json:118]: If this is meant to be fluid, and 1rem is the size, should the min be 16px? Now both min and max are 18px which I believe may just be a typo?

**Comment by kafleg** [comment]
I checked this PR with @michellebulloch in a call and showed her how to test the PR. She is interested to learn and contribute. :)

File: theme.json [hunk 3]
```diff
@@ -97,6 +97,63 @@
...
 			"units": ["%", "px", "em", "rem", "vh", "vw"]
 		},
+		"typography": {
+			"defaultFontSizes": false,
+			"fluid": true,
+			"fontSizes": [
+				{
+					"fluid": false,
+					"name": "Extra Small",
+					"size": "14px",
```
Comment (juanfra) [inline theme.json]: We may want to use `rem` for consistency.

```suggestion
					"size": "0.875rem",
```

File: theme.json [hunk 4]
```diff
@@ -97,6 +97,63 @@
...
+					"slug": "x-small"
+				},
+				{
+					"fluid": {
+						"max": "18px",
+						"min": "18px"
```
Comment (juanfra) [inline theme.json]: We may also want to use `rem` for consistency for the fluid values.

```suggestion
						"max": "1.125rem",
						"min": "1rem"
```

File: theme.json (hunk 2 above)
Comment (juanfra) [reply theme.json:118]: Sounds like it.

File: theme.json [hunk 5]
```diff
@@ -97,6 +97,63 @@
...
+					},
+					"name": "Small",
+					"size": "1rem",
+					"slug": "small"
+				},
+				{
+					"fluid": {
+						"max": "22px",
+						"min": "18px"
```
Comment (juanfra) [inline theme.json]: If we go with `rem`, this should be:

```suggestion
						"max": "1.375rem",
						"min": "1.125rem"
```

File: theme.json [hunk 6]
```diff
@@ -97,6 +97,63 @@
...
+					},
+					"name": "Medium",
+					"size": "1.38rem",
+					"slug": "medium"
+				},
+				{
+					"fluid": {
+						"max": "32px",
+						"min": "28px"
```
Comment (juanfra) [inline theme.json]: If we go with `rem` this should be:

```suggestion
						"max": "2rem",
						"min": "1.75rem"
```

File: theme.json [hunk 7]
```diff
@@ -97,6 +97,63 @@
...
+					},
+					"name": "Large",
+					"size": "1.75rem",
+					"slug": "large"
+				},
+				{
+					"fluid": {
+						"max": "42px",
+						"min": "38px"
```
Comment (juanfra) [inline theme.json]: If we go with `rem`, this should be:

```suggestion
						"max": "2.625rem",
						"min": "2.375rem"
```

File: theme.json [hunk 8]
```diff
@@ -97,6 +97,63 @@
...
+					},
+					"name": "Extra Large",
+					"size": "3rem",
```
Comment (juanfra) [inline theme.json:143]: This looks inconsistent with the min/max defined for XL (`3rem` is `48px`, where the min/max are `38px` and `42px` - In the figma the size is `38px` for XL)

File: theme.json [hunk 9]
```diff
@@ -97,6 +97,63 @@
...
+					"slug": "x-large"
+				},
+				{
+					"fluid": {
+						"max": "58px",
+						"min": "42px"
```
Comment (juanfra) [inline theme.json]: If we go with `rem`, this should be:

```suggestion
						"max": "3.625rem",
						"min": "2.625rem"
```

File: theme.json [hunk 10]
```diff
@@ -97,6 +97,63 @@
...
+					},
+					"name": "Extra Extra Large",
+					"size": "42px",
```
Comment (juanfra) [inline theme.json]:
```suggestion
					"size": "2.625rem",
```

File: theme.json [hunk 11]
```diff
@@ -110,6 +167,125 @@
...
+			},
+			"caption": {
+				"typography": {
+					"fontSize": "var:preset|font-size|x-small",
+					"lineHeight": "1.4"
+				}
+			},
+			"h1": {
+				"typography": {
+					"fontSize": "70px"
```
Comment (juanfra) [inline theme.json:252]: Do we want to make this fluid?

File: theme.json [hunk 12]
```diff
@@ -110,6 +167,125 @@
...
+				}
+			},
+			"h2": {
+				"typography": {
+					"fontSize": "48px"
```
Comment (juanfra) [inline theme.json:257]: Do we want to make this fluid?

**Review by juanfra — COMMENTED** [review]
Thanks so much for working on this, Carolina! I left a few comments, most of them are related to the use of sizes in `rem`.

File: theme.json (hunk 8 above)
Comment (beafialho) [reply theme.json:143]: @juanfra this is from the zip I added to the PR (I set the size presets in the editor). Some of them might not coincide exactly with Figma.

Comment (juanfra) [reply theme.json:143]: Ah ok, thanks for the clarification @beafialho

File: theme.json [hunk 13]
```diff
@@ -97,6 +97,63 @@
...
+						"max": "2rem",
+						"min": "1.75rem"
+					},
+					"name": "Large",
+					"size": "1.75rem",
+					"slug": "large"
+				},
+				{
+					"fluid": {
+						"max": "2.625rem",
```
Comment (carolinan) [inline theme.json]:
```suggestion
						"max": "3rem",
```

**Review by juanfra — APPROVED** [review]
Thanks Carolina! This looks good 🏅 

Let's merge it and we can iterate from here, if there's anything to be adjusted.

**Pull request merged**